pallets/dca/src/lib.rs

```diff
@@ -911,13 +915,30 @@ where
 	fn find_next_free_block(blocknumber: T::BlockNumber) -> Result<T::BlockNumber, DispatchError> {
 		let mut next_execution_block = blocknumber;
 
-		// In a bound fashion, we search for next free block with the delays of 1 - 2 - 4 - 8 - 16.
-		for retry_index in 0u32..=RETRY_TO_SEARCH_FOR_FREE_BLOCK {
+		let mut randomness_generator =
```

Comment:
don't create new instance of randomenes generator, reuse the one you already have!

Reply:
besides like this you will generate exactly same numbers for every schedule in the block...